core/common/assets/js/api/core/commands.js

```diff
@@ -228,22 +230,37 @@ export default class Commands extends elementorModules.Module {
 		return this.currentTrace[ 0 ];
 	}
 
+	validateRun( command, args = {} ) {
+		if ( ! this.commands[ command ] ) {
+			this.error( `\`${ command }\` not found.` );
+		}
+
+		return this.getComponent( command ).dependency( command, args );
+	}
+
 	/**
 	 * Function beforeRun().
 	 *
 	 * @param {string} command
 	 * @param {} args
-	 *
-	 * @returns {boolean} dependency result
 	 */
 	beforeRun( command, args = {} ) {
-		if ( ! this.commands[ command ] ) {
-			this.error( `\`${ command }\` not found.` );
-		}
+		const component = this.getComponent( command ),
+			container = component.getRootContainer();
 
 		this.currentTrace.push( command );
 
-		return this.getComponent( command ).dependency( command, args );
+		Commands.trace.push( command );
+		//Commands.trace.push( this.commands[ command ] );
+
+		this.current[ container ] = command;
+		this.currentArgs[ container ] = args;
+
+		if ( args.onBefore ) {
+			args.onBefore.apply( component, [ args ] );
+		}
+
+		this.trigger( 'run', component, command, args );
```

Comment:
It's updated code? what about `run:before`?

Reply:
run before comes from here:
https://github.com/elementor/elementor/pull/10851